plugins/hashsum_download/server/__init__.py

```diff
@@ -88,4 +128,6 @@ def _getFirstFileByHash(self, algo, hash, user=None):
 
 
 def load(info):
-    HashedFile(info['apiRoot'])
+    HashedFile(info['apiRoot'].file)
+    ModelImporter.model('file').exposeFields(
+        level=AccessType.READ, fields=HashedFile.supportedAlgorithms)
```

Comment:
Note, this means that as future hash types are added, they'll be tacked on to the root of file objects, as opposed to nested in some sub-field. This isn't really a problem though.